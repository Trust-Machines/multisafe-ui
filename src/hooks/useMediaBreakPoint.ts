import {useTheme} from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';

const useMediaBreakPoint = (): [boolean] => {
    const theme = useTheme();
    const isMd = useMediaQuery(theme.breakpoints.up('md'));
    return [isMd];
}

export default useMediaBreakPoint;