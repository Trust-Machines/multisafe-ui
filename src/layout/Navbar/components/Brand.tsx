import React from 'react';
import {useNavigate} from '@reach/router';
import Box from '@mui/material/Box';
import {grey} from '@mui/material/colors';
import {useTheme} from '@mui/material';

import useMediaBreakPoint from '../../../hooks/useMediaBreakPoint';



const Brand = () => {
    const theme = useTheme();
    const [isSm] = useMediaBreakPoint();

    return (
        <Box sx={{
            display: 'flex',
            alignItems: 'center',
            flexShrink: 0,
            flexGrow: 0,
            cursor: 'pointer'
        }} onClick={()=>{
            window.location.href = '/';
        }}>
            <Box sx={{
                width: '32px',
                height: '32px',
                borderRadius: '50%',
                overflow: 'hidden',
                marginRight: '8px'
            }}>
                <img src="/logo.jpg" alt="Logo" style={{
                    width: '100%',
                    height: '100%'
                }}/>
            </Box>
            <Box sx={{
                fontSize: '1.3rem',
                fontWeight: 600,
                color: theme.palette.mode === 'light' ? grey[900] : grey[50],
                display: isSm ? 'block' : 'none'
            }}>MultiSafe</Box>
        </Box>
    );
}

export default Brand;