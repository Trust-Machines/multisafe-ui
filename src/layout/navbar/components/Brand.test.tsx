import {render} from '@testing-library/react';

import Brand from './Brand';
import useMediaBreakPoint from '../../../hooks/use-media-break-point';

jest.mock('../../../hooks/use-media-break-point');

test('1 Default render', () => {
    (useMediaBreakPoint as jest.Mock).mockReturnValue([false, true]);
    const rendered = render(<Brand/>);
    expect(rendered.container).toMatchSnapshot()
});

test('2 Hide brand text', () => {
    (useMediaBreakPoint as jest.Mock).mockReturnValue([true, false]);
    const rendered = render(<Brand/>);
    expect(rendered.container).toMatchSnapshot()
});
