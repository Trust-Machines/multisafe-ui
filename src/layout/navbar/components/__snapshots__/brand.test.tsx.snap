// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <div
    class="MuiBox-root css-1r6mtiz"
  >
    <div
      class="MuiBox-root css-wxdvxx"
    >
      <img
        alt="Logo"
        src="/logo.png"
      />
    </div>
    <div
      class="MuiBox-root css-1dx8jfp"
    >
      MultiSafe
    </div>
  </div>
</div>
`;

exports[`2 Hide brand text 1`] = `
<div>
  <div
    class="MuiBox-root css-176ksq4"
  >
    <div
      class="MuiBox-root css-wxdvxx"
    >
      <img
        alt="Logo"
        src="/logo.png"
      />
    </div>
    <div
      class="MuiBox-root css-1dx8jfp"
    >
      MultiSafe
    </div>
  </div>
</div>
`;
